src/core/services/UserService.ts

```diff
@@ -1,17 +1,101 @@
     return addressEnsName;
   }
+
+  /* -------------------------------------------------------------------------- */
+  /*                                NFT Methods                                 */
+  /* -------------------------------------------------------------------------- */
+  public async getNftBalance(address: string): Promise<NftBalances> {
+    const bluePillNftBalance = await this.getBluePillNftBalance(address);
+    const woofyNftBalance = await this.getWoofyNftBalance(address);
+    return { bluePillNftBalance: bluePillNftBalance, woofyNftBalance: woofyNftBalance };
+  }
+
+  public async getBluePillNftBalance(address: string): Promise<number> {
+    const provider = this.web3Provider.getInstanceOf('ethereum');
+    const { BLUEPILLNFT } = this.config.CONTRACT_ADDRESSES;
+    const bluePillNftContract = getContract(BLUEPILLNFT, bluePillNFTAbi, provider);
+    return this.getUserNftContractBalance(address, bluePillNftContract);
+  }
+
+  public async getWoofyNftBalance(address: string): Promise<number> {
+    const provider = this.web3Provider.getInstanceOf('ethereum');
+    const { WOOFYNFT } = this.config.CONTRACT_ADDRESSES;
+    const woofyNftContract = getContract(WOOFYNFT, woofyNFTAbi, provider);
+    return this.getUserNftContractBalance(address, woofyNftContract);
+  }
+
+  private async getUserNftContractBalance(userAddress: string, nftContract: Contract): Promise<number> {
+    const contractAddress = nftContract.address;
+    let starCount;
+    // Woofy doesn't have its own starCount method in contract, so use custom function
+    switch (contractAddress) {
+      case this.config.CONTRACT_ADDRESSES.WOOFYNFT: {
+        starCount = await this.getWoofyStarCount(nftContract);
+        break;
+      }
+      case this.config.CONTRACT_ADDRESSES.BLUEPILLNFT: {
+        starCount = await nftContract.starCount();
+        starCount = starCount.toNumber();
+        break;
+      }
+      default:
+        starCount = 1;
+        break;
+    }
+    const nftIDs = this.generateNftIds(starCount);
+    const addressArray = this.generateRepeatedAddresses(starCount, userAddress);
+    const balanceArray = await nftContract.balanceOfBatch(addressArray, nftIDs);
+    return this.calculateNftBalanceFromBatch(balanceArray);
+  }
+
+  private async getWoofyStarCount(woofNftContract: Contract): Promise<number> {
+    // Call URI Method with increasing ID's until it fails to find starCount
+    let moreIdsAvailable = true;
+    let knownMax = 2790;
+    while (moreIdsAvailable) {
+      try {
+        await woofNftContract.uri(knownMax);
+        knownMax++;
+      } catch (error) {
+        moreIdsAvailable = false;
+      }
+    }
+    return knownMax - 1;
+  }
+
+  private generateNftIds(starCount: number): number[] {
+    return Array.from({ length: starCount }, (_, i) => i + 1);
+  }
+
+  private generateRepeatedAddresses(starCount: number, address: string): string[] {
+    return Array(starCount).fill(address);
```

Comment:
Would also require `Array(starCount+1).fill(address)` now that we are including the zero base index